app/cdap/components/Connections/Browser/GenericBrowser/index.tsx

```diff
@@ -0,0 +1,156 @@
+import { humanReadableDate, isNilOrEmptyString, objectQuery } from 'services/helpers';
+import { exploreConnection } from 'components/Connections/Browser/GenericBrowser/apiHelpers';
+import * as React from 'react';
+import capitalize from 'lodash/capitalize';
+import If from 'components/If';
+import FolderIcon from '@material-ui/icons/Folder';
+import makeStyle from '@material-ui/core/styles/makeStyles';
+import Table from 'components/Table';
+import TableHeader from 'components/Table/TableHeader';
+import TableRow from 'components/Table/TableRow';
+import TableCell from 'components/Table/TableCell';
+import TableBody from 'components/Table/TableBody';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import { useLocation } from 'react-router-dom';
+
+const useStyle = makeStyle(() => {
+  return {
+    nameWrapper: {
+      display: 'flex',
+      gap: '5px',
+    },
+    grid: {
+      height: '100%',
+      marginTop: '15px',
+    },
+    loadingContainer: {
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      height: '100%',
+    },
+  };
+});
+
+export function GenericBrowser({ selectedConnection }) {
+  const loc = useLocation();
+  const queryParams = new URLSearchParams(loc.search);
+  const pathFromUrl = queryParams.get('path') || '/';
+  const [entities, setEntities] = React.useState([]);
+  const [loading, setLoading] = React.useState(true);
+  const [error, setError] = React.useState(null);
+  const [path, setPath] = React.useState(pathFromUrl);
+  const classes = useStyle();
+
+  const fetchEntities = async () => {
+    try {
+      const res = await exploreConnection({
+        connectionid: selectedConnection,
+        path,
+      });
+      setEntities(res.entities);
+    } catch (e) {
+      setError(`Failed to explore connection : "${e.message}"`);
+    } finally {
+      setLoading(false);
+    }
+  };
+  const onExplore = (entityName) => {
+    if (path === '/') {
+      setPath(`/${entityName}`);
+    } else {
+      setPath(`${path}/${entityName}`);
+    }
+    setLoading(true);
+  };
+  React.useEffect(() => {
+    if (isNilOrEmptyString(selectedConnection)) {
+      return setLoading(false);
+    }
+    fetchEntities();
+  }, [selectedConnection, path]);
+
+  React.useEffect(() => {
+    const query = new URLSearchParams(loc.search);
+    const urlPath = query.get('path') || '/';
+    if (path !== urlPath) {
+      setPath(urlPath);
+    }
+  }, [loc]);
+  if (loading) {
+    return (
+      <div className={classes.loadingContainer}>
+        <LoadingSVG />
+      </div>
+    );
+  }
+
+  if (error) {
+    throw error;
+  }
+  if (!Array.isArray(entities) || (Array.isArray(entities) && !entities.length)) {
+    return <div>No entities available</div>;
+  }
+  let headers = ['Name', 'Type'];
+  const properties = objectQuery(entities, 0, 'properties');
+  headers = [...headers, ...properties.map((prop) => capitalize(prop.key))];
```

Comment:
is it possible that there is no `entities` (ie. empty folder), the `...properties.map` will fail in that case